### PR TITLE
changes to support reset of ttb

### LINKED ISSUE #379: TTB Metadata persists even after updating the key with a TTL
**Describe the bug**

A key set with a TTB has to be deleted before a TTL can be set.. Sample below the updated key is still only visible after 60seconds

**To Reproduce**
1. Create a key with a TTB and then update that key with a TTL 
2. The TTB persists and you have to delete the key and then set the TTL
3. Looks like a secondary code problem as the atClient sends the right metadata
4. 

**Expected behavior**
Old meta data should be ignored when the key is set

**Screenshots**
''' 
var metaData = Metadata()
    ..isPublic = true
    ..isEncrypted = false
    ..namespaceAware = true
    // // one minute
     ..ttb = 60000 ;

  var key = AtKey()
    ..key = hamradio.radioName
    ..sharedBy = currentAtsign
    ..sharedWith = null
    ..metadata = metaData;
;
  await atClient.put(key, message)

var metaData = Metadata()
    ..isPublic = true
    ..isEncrypted = false
    ..namespaceAware = true
    // // one minute
     ..ttl = 60000 ;

  var key = AtKey()
    ..key = hamradio.radioName
    ..sharedBy = currentAtsign
    ..sharedWith = null
    ..metadata = metaData;
;
 /// await atClient.delete(key); <- This line should not have to be used
  await atClient.put(key, message)
'''
**Additional context**

### PR DESCRIPTION
**Pull request by srieteja**
**- What I did**
Added scrutiny for setting ttl and ttb. Added test cases to verify ttb reset. Modifications to functionality and tests to reflect and support changes made in at_utils(https://github.com/atsign-foundation/at_tools/pull/137). Resolves https://github.com/atsign-foundation/at_server/issues/379.

**- How I did it**
Added conditions in if-clauses that are involved in setting ttl/ttb. Modifications done to tests as required to be able to verify changes made and functionality added.

**- How to verify it**
Unit tests are present in 'update_verb_test.dart'.

**- Description for the changelog**

### CODE REVIEW

**Comment by gkc** [comment]
@srieteja would you mind expanding on the description of this pull request ("What I did ...." etc above) please?

i.e. describe what the intent is (e.g. what problem is being addressed) ... "Changes to resolve reset of ttb and ttl" is not really enough detail for me to fully understand the context. Linking the underlying issue(s) e.g. #379 in the summary is usually very helpful

If possible please also summarize the changes being made - or at least, the most important ones. This is especially useful when there are a lot of files to compare and again, helps the reviewers

**Review by gkc — APPROVED** [review]
LGTM

**Comment by gkc** [comment]
@srieteja  Please don't merge until one of @VJag @murali-shris @kalluriramkumar have reviewed also, thanks!

File: at_secondary/at_secondary_server/pubspec.yaml
Comment (murali-shris) [inline at_secondary/at_secondary_server/pubspec.yaml]: at tools changes have to be merged to trunk before merging the server PR

**Comment by murali-shris** [comment]
unit tests are failing without dependency overrides. merging this PR and then will publish at_persistence_secondary

**Pull request merged**
